### PR TITLE
fix(preact-generator): Picking a inbuilt index template and injecting custom tags

### LINKED ISSUE #342: Preact project does not use the generated index.html
While testing out the preact project, I noticed that the roboto font and the title are not set in the running project. then I looked at the build task in preact and it is creating a new file, not using the existing one. In React this works, values set in index.html are preserved when running the app

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**
…jecting custom tags

closes #342

### CODE REVIEW

File: packages/teleport-project-generator-preact/src/constants.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export const customHeadContent = () => `<% if (htmlWebpackPlugin.options.manifest.theme_color) { %>
```
Comment (alexnm) [inline packages/teleport-project-generator-preact/src/constants.ts]: you can keep these as strings if they don't have any dynamic parts

File: packages/teleport-project-generator-preact/src/index.ts [hunk 2]
```diff
@@ -49,6 +47,21 @@ export const createPreactProjectGenerator = () => {
...
       path: ['src'],
       fileName: 'index',
+      options: {
+        appRootOverride: customBodyContent(),
+        customHeadContent: customHeadContent(),
+        customScriptTags: [
+          {
+            target: 'body',
+            type: 'defer',
+            path: `<%= htmlWebpackPlugin.files.chunks['bundle'].entry %>`,
```
Comment (alexnm) [inline packages/teleport-project-generator-preact/src/index.ts]: isn't this the root actually? and this:
```
<%= htmlWebpackPlugin.options.ssr({
	url: '/'
}) %>
```

would be another custom script tag?

File: packages/teleport-project-generator/src/file-handlers.ts [hunk 3]
```diff
@@ -119,18 +127,31 @@ const createHTMLEntryFileChunks = (uidl: ProjectUIDL, options: EntryFileOptions)
...
+
+      const targetNode = target === 'body' ? bodyNode : headNode
       const scriptTag = createHTMLNode('script')
-      if (type === 'module') {
-        addAttributeToNode(scriptTag, 'type', type)
-      } else {
-        addAttributeToNode(scriptTag, 'nomodule', '')
+
+      if (type) {
+        type === 'module'
```
Comment (alexnm) [inline packages/teleport-project-generator/src/file-handlers.ts]: can you please explain the logic here in a comment?

Comment (JayaKrishnaNamburu) [reply packages/teleport-project-generator/src/file-handlers.ts]: This needs a small refactor, the current logic won't work for all edge cases. I will update this now 👍

File: packages/teleport-project-generator-preact/src/index.ts (hunk 2 above)
Comment (JayaKrishnaNamburu) [reply packages/teleport-project-generator-preact/src/index.ts]: Yes, the SSR tag is outside the script and this is added inside a script tags

**Pull request merged**